src/composables/transactionItemList.ts

```diff
@@ -0,0 +1,146 @@
+import { computed } from '@vue/composition-api';
+import {
+  IActiveMultisigTx,
+  IDefaultComposableOptions,
+  ITokenList,
+  ITransaction,
+  ITx,
+} from '../types';
+import {
+  AETERNITY_CONTRACT_ID,
+  defaultTransactionSortingCallback,
+  getInnerTransaction,
+  pipe,
+  TX_FUNCTIONS,
+} from '../popup/utils';
+import { usePendingMultisigTransaction } from './pendingMultisigTransaction';
+import { useTransactionAndTokenFilter } from './transactionAndTokenFilter';
+import { i18n } from '../store/plugins/languages';
+import { useDirectionOfTransaction } from './directionOfTransaction';
+import { useDex } from './dex';
+
+interface TransactionItemList extends IDefaultComposableOptions {
+  isMultisig?: boolean,
+  token: string,
+}
+
+type ITransactionElement = ITransaction | IActiveMultisigTx
+
+export function useTransactionList({
+  store,
+  isMultisig,
+  token,
+}: TransactionItemList) {
+  const availableTokens = computed<ITokenList>(() => store.state.fungibleTokens.availableTokens);
+  const transactions = computed(() => store.state.transactions);
+  const getTxSymbol = computed(() => store.getters.getTxSymbol);
+  const getAccountPendingTransactions = computed<IActiveMultisigTx[]>(
+    () => store.getters.getAccountPendingTransactions,
+  );
+
+  const canLoadMore = computed(() => !!transactions.value.nextPageUrl);
+
+  const {
+    searchPhrase,
+    displayMode,
+    FILTER_MODE,
+  } = useTransactionAndTokenFilter();
+
+  const { pendingMultisigTransaction } = usePendingMultisigTransaction({ store });
+
+  function isFungibleTokenTx(tx: ITx) {
+    return Object.keys(availableTokens.value).includes(tx.contractId);
+  }
+
+  function filterProperTransactions(transactionList: ITransactionElement[]) {
+    return transactionList.filter((transaction) => {
+      const innerTx = getInnerTransaction(transaction.tx!);
+      return !token || (
+        token !== AETERNITY_CONTRACT_ID
+          ? innerTx?.contractId === token
+          : (!innerTx.contractId || !isFungibleTokenTx(innerTx))
+      );
```

Comment:
For better readability I'd suggest not to mix OR operator with the ternary operator